src/app/Core/adaptFilterFromAPI.service.ts

```diff
@@ -14,7 +14,7 @@ export function adaptFilterFromApiService(
             id,
             name,
             parent,
-            values: values.join().split(','),
+            values: (values as Array<string|Date>).map((element: string | Date) => String(element)),

```

Comment:
i thought of adding a comment to use map instead, now you've done it yourself :)
no need to type it here, i think you typed it already in FilterConfigDTO and ClientFilterConfig. So just `values: values.map(value=>String(value))` would do it.

Reply:
If I don't type the values as Array I get the following Error: 

This expression is not callable.
  Each member of the union type '(<U>(callbackfn: (value: string, index: number, array: string[]) => U, thisArg?: any) => U[]) | (<U>(callbackfn: (value: Date, index: number, array: Date[]) => U, thisArg?: any) => U[])' has signatures, but none of those signatures are compatible with each other.

Reply:
I see, typescript generates two mapping function types, so this will not work. But you should be able to simplify to:
```values.map((element: string | Data) => ...)``

Reply:
No, I have already tried that. I need to type values as an Array.

Reply:
Ok, that's strange. Then leave it as it is. In general you would do this also during api data validation and mapping:
````
if(typeof apiRawData === 'Date') {
  myData = parseFromDate(apiRawData)
}
```
Then you do not have to type the api objects before hand, because all you really get from the api is of type unknown and you only tell the compiler how you expect it to look like.
But as mentioned i wonder if you not could tell your json parser to do the conversions for you.